src/contexts/W3iContext/hooks/pushHooks.ts

```diff
@@ -112,5 +118,32 @@ export const usePushState = (w3iProxy: Web3InboxProxy, proxyReady: boolean) => {
     }
   }, [pushClient, refreshPushState])
 
+  useEffect(() => {
+    if (!pushClient) {
+      return noop
+    }
+
+    const pushMessageSub = pushClient.observe('push_message', {
+      next: message => {
+        if (message.params.message.url !== dappOrigin) {
```

Comment:
Curious: do we need this because this observer is also in place in non-dappOrigin scenarios? Or why the safeguard if the client should only be subscribed to push topics relevant to this dapp?

Reply:
Would be great to add a comment explaining intent on top either way 🙏 

There's a lot going on between w3i standalone + the dapp widget integration so we should aim to make the context clearer over time for others regarding why specific checks and/or sub-routines are in place.

Reply:
Great description of the why, ty 🙏